Common/Data/Custom/TradingEconomics/TradingEconomicsCalendar.cs

```diff
@@ -250,6 +374,137 @@ public override DateTimeZone DataTimeZone()
+            var newFigure = value.Replace("$", "")
+                .Replace("(P)", "")
+                .Replace("(R)", "")
+                .Replace("--", "-")
+                .Replace(".5.1", ".5")
+                .Replace("-1-", "-")
+                .Replace("-2-", "-")
+                .Replace("-3-", "-")
+                .Replace("-4-", "-")
+                .Replace("-5-", "-")
+                .Replace("-6-", "-")
+                .Replace("-7-", "-")
+                .Replace("-8-", "-")
+                .Replace("-9-", "-");
+
+            if (newFigure.EndsWith("."))
+            {
+                newFigure = newFigure.Substring(0, newFigure.Length - 1);
+            }
+
+            var inTrillions = newFigure.EndsWith("T");
+            var inBillions = newFigure.EndsWith("B");
+            var inMillions = newFigure.EndsWith("M");
+            var inThousands = newFigure.EndsWith("K");
+
+            // Finally, remove any alphabetical characters from the string before we parse
+            newFigure = Regex.Replace(newFigure, "[^0-9.+-]", "");
+
+            while (Regex.IsMatch(newFigure, @"(\.[0-9]+\.)"))
+            {
+                newFigure = newFigure.Substring(0, newFigure.Length - 1);
+            }
+
+            if (string.IsNullOrWhiteSpace(newFigure))
+            {
+                // U.S. Presidential election is unparsable as decimal.
+                // Other events similar to it might exist as well.
+                return null;
+            }
+
+            // Return null If we can't parse the result as is
+            decimal finalFigure;
+            if (!decimal.TryParse(newFigure, NumberStyles.Any, CultureInfo.InvariantCulture, out finalFigure))
+            {
+                Log.Error($"TradingEconomicsDownloader.ParseDecimal(): Failed to parse the figure {value}. Final form before parsing: {newFigure}");
+                return null;
+            }
+
+            if (inPercent)
+            {
+                return finalFigure / 100m;
+            }
+            if (inTrillions)
+            {
+                return finalFigure * 1000000000000m;
+            }
+            if (inBillions)
+            {
+                return finalFigure * 1000000000m;
+            }
+            if (inMillions)
+            {
+                return finalFigure * 1000000m;
+            }
+            if (inThousands)
+            {
+                return finalFigure * 1000m;
+            }
+
+            return finalFigure;
+        }
+
+        /// <summary>
+        /// Converts country name to currency code (ISO 4217)
+        /// </summary>
+        /// <param name="country">Country name</param>
+        /// <returns>ISO 4217 currency code</returns>
+        public static string CountryToCurrencyCode(string country)
+        {
+            var regions = CultureInfo.GetCultures(CultureTypes.SpecificCultures).Select(x => new RegionInfo(x.LCID));
```

Comment:
It would be nice to have some test for `CountryToCurrencyCode` too. We are performing a `Contains` operation which is case sensitive, should it be insensitive instead? `FirstOrDefault` could return null, `ISOCurrencySymbol` will cause null ref exception. _Maybe_ we would want to keep regions of line 505 in a static collection for performance 🤔